### PR TITLE
improve searching scheduled jobs typos, autoformat, toList

### PR DESCRIPTION
**Pull request by ymo-sci**
- fixes: SIRI-627

### CODE REVIEW

File: src/main/java/sirius/biz/jobs/BasicJobFactory.java
Comment (cschierle) [inline src/main/java/sirius/biz/jobs/BasicJobFactory.java:119]: This seems like a tiny change, but it might be dangerous and breaking. In contrast to `Collectors.toList()`, `Stream.toList()` returns an unmodifiable list.

Aren't there usages, where something is added to the required permissions are where they are overridden?

Same thought goes to the other `toList()` changes of this PR. Please make sure that the return value is not modified elsewhere.

**Review by cschierle — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/jobs/BasicJobFactory.java
Comment (qw3ry) [reply src/main/java/sirius/biz/jobs/BasicJobFactory.java:119]: Please double-check, I remember problems with `toList()`-refactoring when working on tycho jobs in august. Might have been another place, though.

**Review by qw3ry — APPROVED** [review]
If the`.toList()`-refactorings are fine

File: src/main/java/sirius/biz/jobs/BasicJobFactory.java
Comment (mkeckmkeck) [reply src/main/java/sirius/biz/jobs/BasicJobFactory.java:119]: Change to explicit mutable List implementation, or break it and fix all usages and document as immutable

**Review by andyHa — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/biz/jobs/BasicJobFactory.java
Comment (andyHa) [reply src/main/java/sirius/biz/jobs/BasicJobFactory.java:119]: We should now be good to go - after the changes are merged....

**Review by sabieber — APPROVED** [review]

**Review by cschierle — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**
